browser/brave_wallet/ethereum_provider_impl_unittest.cc

```diff
@@ -2171,6 +2180,95 @@ TEST_F(EthereumProviderImplUnitTest, EthSubscribeLogs) {
   EXPECT_FALSE(provider_->eth_logs_tracker_.IsRunning());
 }
 
+TEST_F(EthereumProviderImplUnitTest, EthSubscribeLogsFiltered) {
+  CreateWallet();
+  url_loader_factory_.SetInterceptor(
+      base::BindLambdaForTesting([&](const network::ResourceRequest& request) {
+        url_loader_factory_.ClearResponses();
+
+        std::string header_value;
+        EXPECT_TRUE(request.headers.GetHeader("X-Eth-Method", &header_value));
+
+        std::string content;
+        if (header_value == "eth_getLogs") {
+          auto payload = ToValue(request);
+          EXPECT_TRUE(payload && payload->is_dict());
+
+          auto* method = payload->GetDict().FindString("method");
+          EXPECT_TRUE(method);
+          EXPECT_EQ(*method, "eth_getLogs");
+
+          auto* params = payload->GetDict().FindList("params");
+          EXPECT_TRUE(params);
+          EXPECT_TRUE(params->front().is_dict());
+
+          auto* address = params->front().GetDict().FindList("address");
+          EXPECT_TRUE(address);
+          auto addr = std::find(address->begin(), address->end(), "0x9125");
+          EXPECT_TRUE(addr != address->end());
+
+          auto* fromBlock = params->front().GetDict().FindString("fromBlock");
+          EXPECT_TRUE(fromBlock);
+          EXPECT_EQ(*fromBlock, "0x2211");
+
+          auto* toBlock = params->front().GetDict().FindString("toBlock");
+          EXPECT_TRUE(toBlock);
+          EXPECT_EQ(*toBlock, "0xab65");
+
+          auto* topics = params->front().GetDict().FindList("topics");
+          EXPECT_TRUE(topics);
+          auto topic1 = std::find(topics->begin(), topics->end(), "0x2edc");
+          EXPECT_TRUE(topic1 != topics->end());
+          auto topic2 = std::find(topics->begin(), topics->end(), "0xb832");
+          EXPECT_TRUE(topic2 != topics->end());
+          auto topic3 = std::find(topics->begin(), topics->end(), "0x8dc8");
+          EXPECT_TRUE(topic3 != topics->end());
```

Comment:
Worth just parsing hardcoded json string to base::Value to compare with payload

Reply:
fixed